debian/postinst

```diff
@@ -22,6 +22,31 @@ case $1 in
 	systemctl enable systemd-networkd.service
 
 	systemctl disable walinuxagent.service
+
+	if ! id -u postgres > /dev/null; then
+		# When installing postgres, a postgres user is created unless it
+		# already exists. To have a consistent UID accross installations
+		# and to facilitate migration from Illumos we create a postgres
+		# user with the same UID as in Illumos. The Ubuntu postgres
+		# package requires that the postgres user is also part of the
+		# postgres group. Again, for consistency, we create the postgres
+		# group with a pre-determined GID, which the same as the
+		# postgres UID.
+		addgroup postgres --gid 65437
+		adduser --home /var/lib/postgresql --no-create-home \
+		    --shell /bin/bash --ingroup postgres \
+		    --gecos "PostgreSQL administrator" --uid 65437 \
+		    --disabled-password postgres
+		# To facilitate migration, we also add postgres to group 10,
+		# which is the GID of group staff on Illumos. domain0/mds is
+		# owned by delphix:staff on Illumos.
+		usermod -a -G 10 postgres
```

Comment:
is the plan to eventually remove this?

it's not clear to me if we want all files to use GID 10, or GID "linux staff" after a migration.

Reply:
Yes, the plan would be to remove this some time after the migration has been completed and files/datasets have been chowned appropriately. I agree that there is no value in keeping this around forever.

Reply:
Okay, I think this logic belongs more into the upcoming `delphix-migration` service. This way we won't have postgres beloging to group 10 or 50 for fresh Linux installs.